src/NuGet.DependencyResolver.Core/GraphModel/AnalyzeResult.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace NuGet.DependencyResolver
+{
+    public class AnalyzeResult<TItem>
+    {
+        public List<Tuple<GraphNode<TItem>, GraphNode<TItem>>> Downgrades { get; private set; }
```

Comment:
Nevermind. `Analyze` is generic. One of these days you have to explain why it's so.